### ISSUE TITLE
Go runtime path detection failed on MSYS2

### ISSUE THREAD

**Issue by ipsusila** [Microsoft/vscode-go#983]
On MSYS2, Go binary is not located under `${GOROOT}/bin` thus `getGoRuntimePath()` in `goPath.ts` failed to locate `go.exe`. Example `go env` output in MSYS2:

```
GOROOT=<installation directory>\mingw64\lib\go
GOTOOLDIR=<installation directory>\mingw64\lib\go\pkg\tool\windows_amd64

//Go binary at (accessible through ${PATH})
<installation directory>\mingw64\bin
```
where `<installation directory>` is the directory where I installed MSYS2. Currently I fix this issue by changing `goPath.ts` line 79-84 

```
if (process.env['GOROOT']) {
    runtimePathCache = path.join(process.env['GOROOT'], 'bin', correctBinNameGo);
} else if (process.env['PATH']) {
    let pathparts = (<string>process.env.PATH).split(path.delimiter);
    runtimePathCache = pathparts.map(dir => path.join(dir, correctBinNameGo)).filter(candidate => fileExists(candidate))[0];
}
```

to

```
if (process.env['GOROOT']) {
    runtimePathCache = path.join(process.env['GOROOT'], 'bin', correctBinNameGo);
    if (!fileExists(runtimePathCache))
        runtimePathCache = null;
} 
if (!runtimePathCache && process.env['PATH']) {
    let pathparts = (<string>process.env.PATH).split(path.delimiter);
    runtimePathCache = pathparts.map(dir => path.join(dir, correctBinNameGo)).filter(candidate => fileExists(candidate))[0];
}
```

**Comment by ramya-rao-a** [comment]
Thanks for reporting @ipsusila !

I have committed the fix (https://github.com/Microsoft/vscode-go/commit/67557af839921649e9fb1653b704b01b3ced307b) and it will be out in the next update

**Comment by ramya-rao-a** [comment]
The fix is now available in the latest update (0.6.62)
